cibuildwheel/windows.py

```diff
@@ -115,6 +122,50 @@ def install_pypy(tmp: Path, arch: str, url: str) -> Path:
     return installation_path / "python.exe"
 
 
+def setup_setuptools_cross_compile(
+    python_configuration: PythonConfiguration,
+    python_libs_base: Path,
+    env: Dict[str, str],
+) -> None:
+    # We write to distutils_cfg for distutils-based builds because we know we
+    # currently don't have one (unless it's our own from a previous build)
+    for p in env["PATH"].split(os.pathsep):
+        distutils_cfg = Path(p) / "Lib/site-packages/setuptools/_distutils/distutils.cfg"
+        if distutils_cfg.parent.is_dir():
+            break
+    else:
+        log.warning("Did not find setuptools install to configure")
+        return
```

Comment:
Ah, this confused me for a while, but here you're locating the virtualenv-installed version of setuptools, and modifying that. That's good, much better than the system-installed version as I initially thought.

Another idea I had around this was to write a [user-level distutils.cfg](https://github.com/pypa/setuptools/blob/641cb8955923f5b23c5be7cdd3edb7427c40995c/setuptools/_distutils/dist.py#L346) at `<tmpdir>/pydistutils.cfg` and set `USERPROFILE=<tmpdir>` so that distutils picks it up. That might help with users that don't use setuptools, and would negate the need for `SETUPTOOLS_USE_DISTUTILS` I think. Any thoughts?

Reply:
I'm afraid we can't negate the `SETUPTOOLS_USE_DISTUTILS` option - there are actual bugfixes in setuptools that are not in the standard library (and never will be).

I also wouldn't want to mess with anything that might be used in an existing build, and my experience of `setup.py` files is that could be _literally anything_. "Redirecting" the user profile[^1] is one such thing that could break builds for unrelated reasons.

I tried a lot of ways to pass the required arguments into setuptools, and this is by far the safest way that doesn't rely on invoking `setup.py` directly. Personally that's how I set up my own builds, but it's not a good option for cibuildwheel.

[^1]: It's actually just pretending that it's been redirected, and using the leakiest possible way to do so. Really not a good idea.

Reply:
Happy to expand the comment to help the next person figure it out quicker though, something like:

```suggestion
    # We can write to distutils.cfg in our venv to enable distutils-based builds
    # because we know we currently don't have one (unless it's our own from a
    # previous build)
    for p in env["PATH"].split(os.pathsep):
        distutils_cfg = Path(p) / "Lib/site-packages/setuptools/_distutils/distutils.cfg"
        if distutils_cfg.parent.is_dir():
            break
    else:
        log.warning("Did not find setuptools install to configure")
        return
```